lib/dialects/abstract/query-generator.js

```diff
@@ -2228,6 +2228,19 @@ const QueryGenerator = {
     return result ? result : '1=1';
   },
 
+  // A recursive parser for nested where conditions
+  parseConditionObject(conditions, path) {
+    path = path || [];
+    return _.reduce(conditions, (r, v, k) => { // result, key, value
```

Comment:
Yes, I just copy-and-pasted existing things without modification. 
I'll change similar things anyway :)